src/components/TodoItem.tsx

```diff
@@ -0,0 +1,66 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Action } from '../enums/Action';
+import { Todo } from '../types/Todo';
+import { TodoEdit } from './TodoEdit';
+
+type Props = {
+  todo: Todo,
+  dispatch: React.Dispatch<{ type: Action, payload?: unknown }>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo: { id, title, completed }, dispatch,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+
+  return (
+    <li
+      data-cy="todo"
+      className={classNames(
+        'todo',
+        { completed },
+      )}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          defaultChecked
+          onClick={() => dispatch({
+            type: Action.UPDATE,
+            payload: [id, { completed: !completed }],
+          })}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)